packages/slice-machine/src/modules/slices/index.ts

```diff
@@ -148,18 +148,7 @@ export const slicesReducer: Reducer<SlicesStoreType | null, SlicesActions> = (
     }
     case getType(pushSliceCreator.success): {
       const newComponentUI = action.payload.component;
-
-      const newRemoteSlices = [...state.remoteSlices];
-
-      const remoteSliceIndex = state.remoteSlices.findIndex(
-        ({ id }) => id === newComponentUI.model.id
-      );
-
-      if (remoteSliceIndex !== -1) {
-        newRemoteSlices[remoteSliceIndex] = newComponentUI.model;
-      } else {
-        newRemoteSlices.push(newComponentUI.model);
-      }
+      const newRemoteSlices = action.payload.remoteSlices;
```

Comment:
why are we changing this code and adding a new payload which requires a new api call with `yield call(getState))`? what was wrong in the old logic?

Reply:
remote slices was not being updated, so when pushing a new slice it was not updated in the remoteSlice property

Reply:
yeah but why? didn't the code `const newRemoteSlices = [...state.remoteSlices];

      const remoteSliceIndex = state.remoteSlices.findIndex(
        ({ id }) => id === newComponentUI.model.id
      );

      if (remoteSliceIndex !== -1) {
        newRemoteSlices[remoteSliceIndex] = newComponentUI.model;
      } else {
        newRemoteSlices.push(newComponentUI.model);
      }` work?

Reply:
I think the actual bug was that we are returning `return { ...state, libraries: newLibraries };` instead of `return { ...state, libraries: newLibraries, remoteSlices: newRemoteSlices };`

we were computing the newRemoteSlices but not returning them.

Reply:
are those remote slices the remote slices in the state before making the request?

I'll see what. i can do

Reply:
done?